codeworld-api/src/CodeWorld/Driver.hs

```diff
@@ -1738,34 +1743,34 @@ highlightDrawing (DrawState at _) drawing =
 
 getDrawNode :: NodeId -> Drawing m -> Maybe (Drawing m, DrawState)
 getDrawNode n _
-    | n < 0 = Nothing
+    | n < (NodeId 0) = Nothing
 getDrawNode n drawing = either Just (const Nothing) $ go initialDS n drawing
   where
-    go ds 0 d = Left (d, ds)
-    go ds n (Shape _) = Right (n - 1)
-    go ds n (Transformation f dr) = go (f ds) (n - 1) dr
-    go ds n (Drawings []) = Right (n - 1)
+    go ds (NodeId 0) d = Left (d, ds)
+    go ds n (Shape _) = Right (pred n)
+    go ds n (Transformation f dr) = go (f ds) (pred n) dr
+    go ds n (Drawings []) = Right (pred n)
     go ds n (Drawings (dr:drs)) =
-        case go ds (n - 1) dr of
+        case go ds (pred n) dr of
             Left d -> Left d
-            Right n -> go ds (n + 1) $ Drawings drs
+            Right n -> go ds (succ n) $ Drawings drs
 
 replaceDrawNode :: forall m. NodeId -> Drawing m -> Drawing m -> Maybe (Drawing m)
 replaceDrawNode n _ _
-    | n < 0 = Nothing
+    | n < (NodeId 0) = Nothing
```

Comment:
I'm not quite sure why we have this case checking for an invalid NodeId. Maybe we should use `error` here?

Reply:
I don't know.  I think the front end code uses -1 to indicate no node is selected, so this might be used.